tensorzero-core/src/error.rs

```diff
@@ -801,6 +805,16 @@ impl ErrorDetails {
             tracing::Level::TRACE => tracing::trace!("{self}"),
         }
     }
+
+    pub fn nonretryable(&self) -> bool {
+        match &self {
+            ErrorDetails::RateLimitExceeded { .. } => true,
+            ErrorDetails::ModelProvidersExhausted { provider_errors } => provider_errors
+                .iter()
+                .all(|(_, error)| error.nonretryable()),
```

Comment:
I don't know the code but will this mean "if any provider is retryable, all providers will be retried" or only the retryable ones will be retried?